_source/_posts/2022-06-09-oauth-java.adoc

```diff
@@ -0,0 +1,334 @@
+        http
+            .authorizeRequests(request -> request.anyRequest().authenticated())
+            .oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
+        return http.build();
+    }
+}
+----
+
+TIP: If you use the https://github.com/okta/okta-spring-boot[Okta Spring Boot Starter], you do not need to configure things with Java. We've included this configuration for you.
+
+=== Watch OAuth 2.0 in action with Java frameworks
+
+I've created a few screencasts to show how to build a REST API with Quarkus, Spring Boot, Micronaut, and Helidon. These videos also show how to secure each API with OAuth 2.0.
+
+++++
+<div class="video-grid">
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/3J9uFVd3BoY" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/w-qKailh3WQ" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/IG2uo4IP1QI" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/Bj69rOIwQwY" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+</div>
+++++
+
+At the end of each video, I show how the https://cli.okta.com[Okta CLI] allows you to create working examples for each framework with the `okta start <framework>` command.
+
+If you prefer reading and copy pasta over video, check out these quick tutorials:
+
+- link:/blog/2022/05/23/java-rest-api-quarkus[Kickstart a Java REST API with Quarkus]
+- link:/blog/2022/05/25/java-rest-api-spring-boot[Start a Java REST API with Spring Boot]
+- link:/blog/2022/05/31/java-rest-api-micronaut[Launch a Java REST API with Micronaut]
+- link:/blog/2022/06/02/java-rest-api-helidon[Create a Java REST API with Helidon]
+
+== Java frameworks without OAuth support
+
+You might get away with saying that JSF and Wicket don't have OAuth support. However, they both build on the Servlet API so it is possible to use them with security frameworks I mentioned earlier.
+
+They might not have a dedicated security plugin for OAuth, but that's not important if you can get _something_ to work, IMHO.
+
+== Java security frameworks
+
+There are a few Java security frameworks that aren't tied to a specific framework.
+
+=== Apache Shiro
+
+Apache Shiro expect to operate in a servlet environment and can work with any framework that's based on servlets and filters. Examples include ...
```

Comment:
@bdemers Is Shiro dependent on servlets and filters? Looking at the [integrations](https://shiro.apache.org/integration.html) page, it looks like it might be capable of a lot more.

Reply:
Mostly, while it is possible use Shiro elsewhere (A while back I used Shiro with Apache MINA for an SSHd integration), the only out-of-the-box web support depends on servlets.

Reply:
What do you think the best integrations to highlight are? A lot of them on the integrations page are old or 404. For example, Stormpath works but probably should be removed. Play 404s. Wicket is 9 years old and Lift is 7 years old.

Reply:
A quick scrub of the page: https://github.com/apache/shiro-site/pull/160/

I haven't tried it out, but the buji-pac4j seems like its been the most popular (and for some reason, it wasn't added in this list, added in the above PR)